src/integer_mod_q/mat_polynomial_ring_zq/get.rs

```diff
@@ -0,0 +1,351 @@
+        let (row_i64, column_i64) = evaluate_indices(self, row, column)?;
+
+        // since `self.matrix` is a correct fmpz_poly matrix and both row and column
+        // are previously checked to be inside of the matrix, no errors
+        // appear inside of `unsafe` and `fmpz_poly_set` can successfully clone the
+        // entry of the matrix. Therefore no memory leaks can appear.
+        let mut copy = PolyOverZ::default();
+        let entry = unsafe { fmpz_poly_mat_entry(&self.matrix.matrix, row_i64, column_i64) };
+        unsafe { fmpz_poly_set(&mut copy.poly, entry) };
+
+        Ok(copy)
+    }
+}
+
+impl GetEntry<PolynomialRingZq> for MatPolynomialRingZq {
+    /// Outputs the [`PolynomialRingZq`] value of a specific matrix entry.
+    ///
+    /// Parameters:
+    /// - `row`: specifies the row in which the entry is located
+    /// - `column`: specifies the column in which the entry is located
+    ///
+    /// Returns the [`PolynomialRingZq`] value of the matrix at the position of the given
+    /// row and column or an error, if the number of rows or columns is
+    /// greater than the matrix or negative.
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer_mod_q::MatPolynomialRingZq;
+    /// use qfall_math::integer_mod_q::ModulusPolynomialRingZq;
+    /// use qfall_math::integer_mod_q::PolynomialRingZq;
+    /// use qfall_math::integer::MatPolyOverZ;
+    /// use qfall_math::integer::PolyOverZ;
+    /// use qfall_math::traits::*;
+    /// use std::str::FromStr;
+    ///
+    /// let modulus = ModulusPolynomialRingZq::from_str("4  1 0 0 1 mod 17").unwrap();
+    /// let poly_mat = MatPolyOverZ::from_str("[[4  -1 0 1 1, 1  42],[0, 2  1 2]]").unwrap();
+    /// let poly_ring_mat = MatPolynomialRingZq::from((&poly_mat, &modulus));
+    ///
+    /// let entry: PolynomialRingZq = poly_ring_mat.get_entry(1,0).unwrap();
+    /// ```
+    ///
+    /// # Errors and Failures
+    /// - Returns a [`MathError`] of type [`OutOfBounds`](MathError::OutOfBounds)
+    /// if the number of rows or columns is greater than the matrix or negative.
+    fn get_entry(
+        &self,
+        row: impl TryInto<i64> + Display,
+        column: impl TryInto<i64> + Display,
+    ) -> Result<PolynomialRingZq, MathError> {
+        let (row_i64, column_i64) = evaluate_indices(self, row, column)?;
+
+        // since `self.matrix` is a correct fmpz_poly matrix and both row and column
+        // are previously checked to be inside of the matrix, no errors
+        // appear inside of `unsafe` and `fmpz_poly_set` can successfully clone the
+        // entry of the matrix. Therefore no memory leaks can appear.
+        let mut copy = PolynomialRingZq {
+            poly: PolyOverZ::default(),
+            modulus: self.get_mod(),
+        };
+        let entry = unsafe { fmpz_poly_mat_entry(&self.matrix.matrix, row_i64, column_i64) };
+        unsafe { fmpz_poly_set(&mut copy.poly.poly, entry) };
+
+        Ok(copy)
```

Comment:
rather use get_entry to get a PolyOverZ and then just make a polynomialRingZq out of it -> reduces lines of codes and reduces redundancies